itou/job_applications/test_notify_pole_emploi.py

```diff
@@ -0,0 +1,144 @@
+import datetime
+from unittest.mock import patch
+
+import httpx
+import respx
+from django.test import TestCase, override_settings
+from django.utils import timezone
+
+from itou.job_applications.factories import JobApplicationWithApprovalFactory
+from itou.job_applications.models import pole_emploi_api_client
+from itou.users.factories import JobSeekerFactory
+from itou.utils.apis.pole_emploi import PoleEmploiApiClient
+from itou.utils.mocks.pole_emploi import (
+    PE_API_MAJPASS_RESULT_ERROR_MOCK,
+    PE_API_MAJPASS_RESULT_OK_MOCK,
+    PE_API_RECHERCHE_MANY_RESULTS_MOCK,
+    PE_API_RECHERCHE_RESULT_KNOWN_MOCK,
+)
+
+
+@override_settings(
+    API_ESD={
+        "BASE_URL": "https://base.domain",
+        "AUTH_BASE_URL": "https://authentication-domain.fr",
+        "KEY": "foobar",
+        "SECRET": "pe-secret",
+    }
+)
+class JobApplicationNotifyPoleEmploiIntegrationTest(TestCase):
+    def setUp(self):
+        self.api_client = PoleEmploiApiClient()
+        respx.post(self.api_client.token_url).mock(
+            return_value=httpx.Response(200, json={"token_type": "foo", "access_token": "batman", "expires_in": 3600})
+        )
+
+    def test_invalid_job_seeker_for_pole_emploi(self):
+        """
+        Error case: our job seeker is not valid (from PoleEmploi’s point of view: here, the NIR is missing)
+         - We do not even call the APIs
+         - no entry should be added to the notification log database
+        """
+        now = timezone.now()
+        job_seeker = JobSeekerFactory(nir="")
+        job_application = JobApplicationWithApprovalFactory(job_seeker=job_seeker)
+        job_application.notify_pole_emploi(at=now)
+        job_application.approval.refresh_from_db()
+        self.assertEqual(job_application.approval.pe_notification_status, "notification_error")
+        self.assertEqual(job_application.approval.pe_notification_time, now)
+        self.assertEqual(job_application.approval.pe_notification_endpoint, "rech_individu")
+        self.assertEqual(job_application.approval.pe_notification_exit_code, "MISSING_DATA")
+
+    @respx.mock
+    def test_notification_accepted_nominal(self):
+        now = timezone.now()
+        respx.post(self.api_client.recherche_individu_url).mock(
+            return_value=httpx.Response(200, json=PE_API_RECHERCHE_RESULT_KNOWN_MOCK)
+        )
```

Comment:
Pour les cas généraux, il y a un raccourcis :
```suggestion
        respx.post(self.api_client.recherche_individu_url).respond(200, json=PE_API_RECHERCHE_RESULT_KNOWN_MOCK)
```
https://lundberg.github.io/respx/guide/#shortcuts